### DISCUSSION TITLE
Inquiry Regarding Testing Methodology for Azure Monitor Integration in Vector

### DISCUSSION THREAD

**Discussion by vparfonov** [vectordotdev/vector#19043] [Q&A] [answered]
Hello, Vector dev community
I am currently exploring the integration capability of Vector for forwarding log events to Azure Monitor. In my initial tests, the forwarding functionality appeared to work correctly, providing me with confidence in the basic functionality. As I prepare for the next phase, I am particularly interested in understanding the approach to integration testing within Vector, specifically for Azure Monitor Sink.

I noticed that there is currently no integration test available for Azure Monitor Sink in your code base. Could you please provide more information about your testing methodology for validating the integration of Vector with Azure Monitor? I am keen to learn about your strategies for testing log event transmission, interaction handling, and data validation within Azure Monitor. Any insights into the tools, frameworks, or techniques utilized for testing the interaction with Azure Monitor would be greatly appreciated.

Thanks

**Comment by StephenWakely** [comment, marked as answer]
Hi, you are correct there are no integration tests for the Azure Monitor sinks, this is an area where the project is lacking. We rely on manual testing for any changes to this sink.

To implement integration testing within Vector we rely on there being a Docker image available that can emulate the service we are sending or receiving data to/from. To my knowledge (and I'd be very happy to be corrected here!) no such image exists for Azure Monitor so we are unable to develop these tests.

We would be very happy to take any contributions if someone were willing to develop an emulator for us!

**Reply by vparfonov** [reply]
Thanks a lot for answer
